### PR TITLE
Added ignore case option to StringSearch

### PR DESCRIPTION
**Pull request by FriedDonkey**
Added functionality to ignore case

### CODE REVIEW

File: lab4-new.java/src/StringSearch.java [hunk 1]
```diff
@@ -31,15 +31,21 @@ public static ArrayList<String> search(String word, File file) {
...
+    public static String checkLine(String word, String line, Boolean ignoreCase) {
+        if(ignoreCase && line.toLowerCase().contains(word.toLowerCase())) {
+            return line;
+        } else if(line.contains(word)) {
             return line;
         }
         return null;
     }
 
     public static void main(String[] args) {
```
Comment (ThildaHolmner) [inline lab4-new.java/src/StringSearch.java:70]: Never handles cases with arg.length() == 1

File: lab4-new.java/src/StringSearch.java [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 public class StringSearch {
 
-    public static ArrayList<String> search(String word, File file) {
+    public static ArrayList<String> search(String word, File file, Boolean ignoreCase) {
```
Comment (ThildaHolmner) [inline lab4-new.java/src/StringSearch.java]: Consider using the word "pattern" and not "word" for clarity.

File: lab4-new.java/src/StringSearch.java [hunk 3]
```diff
@@ -31,15 +31,21 @@ public static ArrayList<String> search(String word, File file) {
     }
 
 
-    public static String checkLine(String word, String line) {
-        if(line.contains(word)) {
+    public static String checkLine(String word, String line, Boolean ignoreCase) {
+        if(ignoreCase && line.toLowerCase().contains(word.toLowerCase())) {
```
Comment (ThildaHolmner) [inline lab4-new.java/src/StringSearch.java]: `line.toLowerCase().contains(word.toLowerCase())` has hard readability.

File: lab4-new.java/src/StringSearch.java [hunk 4]
```diff
@@ -31,15 +31,21 @@ public static ArrayList<String> search(String word, File file) {
...
+        } else if(line.contains(word)) {
             return line;
         }
         return null;
     }
 
     public static void main(String[] args) {
-        System.out.println(search(args[0], new File(args[1])));
+        Boolean ignoreCase = false;
+        if(args.length == 3 && args[2].equals("-i")) {
```
Comment (ThildaHolmner) [inline lab4-new.java/src/StringSearch.java]: Provide supportive feedback to user when there is a wrong input.

File: lab4-new.java/src/StringSearch.java [hunk 5]
```diff
@@ -17,7 +17,7 @@ public static ArrayList<String> search(String word, File file) {
 
             while (scanner.hasNextLine()) {
```
Comment (ThildaHolmner) [inline lab4-new.java/src/StringSearch.java:23]: You do not handle error reading in file

File: lab4-new.java/src/StringSearch.java
Comment (ThildaHolmner) [inline lab4-new.java/src/StringSearch.java:1]: Test can be good but not necessary for this project.

Comment (ThildaHolmner) [inline lab4-new.java/src/StringSearch.java:1]: Not an excessive amount of methods. But for clarity, you should add comments describing why the implemented methods are needed.

Comment (ThildaHolmner) [inline lab4-new.java/src/StringSearch.java:1]: You have not formatted the code properly.

File: lab4-new.java/src/StringSearch.java [hunk 6]
```diff
@@ -8,7 +8,7 @@
 
 public class StringSearch {
 
-    public static ArrayList<String> search(String word, File file) {
+    public static ArrayList<String> search(String word, File file, Boolean ignoreCase) {
 
         ArrayList<String> matches = new ArrayList<>();
```
Comment (ThildaHolmner) [inline lab4-new.java/src/StringSearch.java:18]: the name "matches" is unclear

**Review by ThildaHolmner — COMMENTED** [review]
Well done! We have made some comments for you to look at

File: lab4-new.java/src/StringSearch.java (hunk 1 above)
Comment (signejohansson) [reply lab4-new.java/src/StringSearch.java:70]: Added a method for checking correct number of arguments

File: lab4-new.java/src/StringSearch.java (hunk 2 above)
Comment (signejohansson) [reply lab4-new.java/src/StringSearch.java]: Fixed, changed it

File: lab4-new.java/src/StringSearch.java (hunk 3 above)
Comment (signejohansson) [reply lab4-new.java/src/StringSearch.java]: Fixed, divided it into local variables with descriptive names instead

File: lab4-new.java/src/StringSearch.java (hunk 4 above)
Comment (signejohansson) [reply lab4-new.java/src/StringSearch.java]: Fixed

File: lab4-new.java/src/StringSearch.java (hunk 5 above)
Comment (signejohansson) [reply lab4-new.java/src/StringSearch.java:23]: We think that we do handle this with a try-catch, but we have changed to print an error message also.

File: lab4-new.java/src/StringSearch.java
Comment (signejohansson) [reply lab4-new.java/src/StringSearch.java:1]: Added javadoc comments to methods

Comment (signejohansson) [reply lab4-new.java/src/StringSearch.java:1]: Done

Comment (signejohansson) [reply lab4-new.java/src/StringSearch.java:1]: Yes, that is reasonable

File: lab4-new.java/src/StringSearch.java (hunk 6 above)
Comment (signejohansson) [reply lab4-new.java/src/StringSearch.java:18]: We disagree, but thanks anyway :)

**Review by ThildaHolmner — APPROVED** [review]
Thanks! I will merge changes now.

**Pull request merged**